designer/client/src/windowManager/LoadingButton.tsx

```diff
@@ -0,0 +1,38 @@
+import { FooterButtonProps } from "@touk/window-manager/cjs/components/window/footer/FooterButton";
+import React, { useCallback, useState } from "react";
+import { LoadingButton as MuiLoadingButton } from "@mui/lab";
+
+export const LoadingButton = (props: FooterButtonProps): JSX.Element => {
+    const { classname, action, title, disabled } = props;
+    const [loading, setLoading] = useState(false);
+    const onClick = useCallback(async () => {
+        setLoading(true);
+        try {
+            await action();
+        } catch (e) {
+            //ignore
+        }
+        setLoading(false);
+    }, [action]);
+
+    return (
+        <MuiLoadingButton
+            color={classname === "secondary-button" || classname === "tertiary-button" ? "inherit" : "primary"}
```

Comment:
i'd like to have some const/enum for such strings.
there are better ways to find usages than plain text search ;)

Reply:
Done ✔️